pkg/executor/coprocessor.go

```diff
@@ -182,14 +182,14 @@ func (h *CoprocessorDAGHandler) buildDAGExecutor(req *coprocessor.Request) (exec
 	}
 
 	stmtCtx := h.sctx.GetSessionVars().StmtCtx
-	stmtCtx.SetFlagsFromPBFlag(dagReq.Flags)
+
 	tz, err := timeutil.ConstructTimeZone(dagReq.TimeZoneName, int(dagReq.TimeZoneOffset))
+	h.sctx.GetSessionVars().TimeZone = tz
```

Comment:
Any reason to put this line before `err != nil` ?

Reply:
No. I moved it under the `err != nil`.